adaptivefiltering/segmentation.py

```diff
@@ -37,16 +37,33 @@ def __geo_interface__(self):
 
 class Segmentation(geojson.FeatureCollection):
     @classmethod
-    def load(cls, filename):
+    def load(cls, filename=None):
         """Load segmentation from a filename
 
         :param filename:
             The filename to load from. Relative paths are interpreted
             w.r.t. the current working directory.
         :type filename: str
         """
-        with open(filename, "r") as f:
-            return Segmentation(geojson.load(f))
+        # calls a widget to upload and directly use data from local machine.
+        if filename == None:
+            from adaptivefiltering.widgets import upload_files
+
+            filename = upload_files(
+                directory="./filters_test_upload/", filetype=".geojson"
+            )
+
+        # if a list of files is given a lost of segmentations will be returned.
+        if isinstance(filename, list):
```

Comment:
It is more pythonic to check against `collections.abc.Iterable` - imagine doing the wrong thing because somebody provided `tuple` instead of `list`.